sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -898,28 +915,40 @@ static int sof_card_dai_links_create(struct device *dev,
 		if (!ssp_components)
 			return -ENOMEM;
 
-		info = &codec_info_list[index];
+		info = &codec_info_list[ssp_codec_index];
 		codec_name = devm_kasprintf(dev, GFP_KERNEL, "i2c-%s:0%d",
 					    info->acpi_id, j++);
 		if (!codec_name)
 			return -ENOMEM;
 
 		ssp_components->name = codec_name;
 		ssp_components->dai_name = info->dai_name;
-		init_dai_link(links, id++, name, info->direction[0],
-			      info->direction[1], cpus, cpu_name,
-			      ssp_components, 1, dai_id++, NULL, info->ops);
+		cpus[cpu_id].dai_name = cpu_name;
+
+		playback = info->direction[SNDRV_PCM_STREAM_PLAYBACK];
+		capture = info->direction[SNDRV_PCM_STREAM_CAPTURE];
+		init_dai_link(links + link_id, name, playback, capture,
+			      cpus + cpu_id, 1, ssp_components, 1, be_id,
```

Comment:
try to split cpus and codec dais (as I suggested in my patch).

It's a lot clearer if you do 

````c
init_dai_link(links + link_id, name, playback, capture,
                cpus + cpu_id, 1, 
                ssp_components, 1, 
               be_id,			     
                NULL, info->ops);
￼````

and use the same format for all calls to init_dai_link,

Please add this in a follow-up PR

Reply:
yes,  I will send another PR for this